resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/submit/KubernetesClientApplication.scala

```diff
@@ -145,6 +145,7 @@ private[spark] class Client(
         s"${resolvedDriverPod.getMetadata.getName}"
       if (waitForAppCompletion) {
         logInfo(s"Waiting for application ${conf.appName} with submission ID ${sId} to finish...")
+        watcher.start()
```

Comment:
Instead of this, I think it would be simpler / better to start the watcher in the constructor. That way, if the log period is not specified, you can even avoid creating the executor for the logging task altogether.